build_tools/azure/install.sh

```diff
@@ -129,9 +125,7 @@ scikit_learn_install() {
         export LDFLAGS="$LDFLAGS -Wl,--sysroot=/"
     fi
 
-    if [[ "$BUILD_WITH_SETUPTOOLS" == "true" ]]; then
-        python setup.py develop
-    elif [[ "$PIP_BUILD_ISOLATION" == "true" ]]; then
+    if [[ "$PIP_BUILD_ISOLATION" == "true" ]]; then
```

Comment:
This means that we never check in the CI that scikit-learn can be built against "old" versions of our build dependencies, like numpy<2. Don't you think we should keep some jobs where we set `--no-build-isolation` ?

Reply:
I think I haven't changed this part :thinking:, maybe it is the diff that is a bit confusing?

Basically I removed the `if [[ "$BUILD_WITH_SETUPTOOLS" == "true" ]]` but there is still the same clause as before `if [[ "$PIP_BUILD_ISOLATION" == "true" ]]` (except that `elif` became `if`).

I have not changed this in this PR but basically all builds except one are using `--no-build-isolation` and some of them are building with numpy<2 (depends on the lock-file or whether they use system Numpy e.g. for the Ubuntu Atlas build).

Reply:
oh right, I misread the diff...